tests/unittests/config/test_cc_keyboard.py

```diff
@@ -73,5 +76,46 @@ def test_schema_validation(self, config, error_msg):
             with pytest.raises(SchemaValidationError, match=error_msg):
                 validate_cloudconfig_schema(config, schema, strict=True)
 
+    @mock.patch("cloudinit.distros.Distro.uses_systemd")
+    @mock.patch("cloudinit.distros.subp.subp")
+    def test_systemd_linux_cmd(self, m_subp, m_uses_systemd, *args):
+        """Ubuntu runs localectl"""
+        cfg = {"keyboard": {"layout": "us", "variant": "us"}}
+        layout = "us"
+        model = "pc105"
+        variant = "us"
+        options = ""
+        m_uses_systemd.return_value = True
+        cloud = get_cloud("ubuntu")
+        cc_keyboard.handle("cc_keyboard", cfg, cloud, [])
+        locale_calls = [
+            mock.call(
+                [
+                    "localectl",
+                    "set-x11-keymap",
+                    layout,
+                    model,
+                    variant,
+                    options,
+                ]
+            ),
+            mock.call(
+                ["systemctl", "restart", "console-setup"],
+                capture=True,
+                rcs=None,
+            ),
+        ]
+        m_subp.assert_has_calls(locale_calls)
+
+    @mock.patch("cloudinit.distros.subp.subp")
+    def test_alpine_linux_cmd(self, m_subp, *args):
+        """Alpine Linux runs setup-keymap"""
+        cfg = {"keyboard": {"layout": "us", "variant": "us"}}
+        layout = "us"
+        variant = "us"
+        cloud = get_cloud("alpine")
+        cc_keyboard.handle("cc_keyboard", cfg, cloud, [])
+        m_subp.assert_called_once_with(["setup-keymap", layout, variant])
+
```

Comment:
Nit: Might be worth another test or two testing errors and ignores, but I'll leave that decision to you.

Reply:
I added an additional testcase,test_alpine_linux_ignore_model.

I wasn't sure how to do testcases for errors as when I tried the error tests then failed.